src/SpendPermissionManager.sol

```diff
@@ -153,14 +154,19 @@ contract SpendPermissionManager is EIP712 {
 
     /// @notice Approve a spend permission via a signature from the account.
     ///
+    /// @dev Compatible with EIP-6492 signatures (https://eips.ethereum.org/EIPS/eip-6492)
+    ///
     /// @param spendPermission Details of the spend permission.
     /// @param signature Signed approval from the user.
     function permit(SpendPermission memory spendPermission, bytes memory signature) public {
-        // validate signature over spend permission data
+        // validate signature over spend permission data, compatible with EIP6492
         if (
-            IERC1271(spendPermission.account).isValidSignature(getHash(spendPermission), signature)
-                != IERC1271.isValidSignature.selector
+            SignatureCheckerLib.isValidERC6492SignatureNowAllowSideEffects(
+                spendPermission.account, getHash(spendPermission), signature
+            )
         ) {
+            _approve(spendPermission);
```

Comment:
This extra approval is extraneous because we approve after this conditional block already! Mind making the conditional check for `!valid` and then revert?